### PR TITLE
gh-99: Fix eslint errors/warns on BE

### PR DESCRIPTION
**Pull request by littlewhywhat**
Closes gh-99

In many cases here are the minimal fixes (to solve eslint problems) with some extra effort to add crucial type-checking.

Definitely there is a room for improvement - the minimal scope I chose for this issue was to resolve the eslint errors and warnings.

Maybe we could add type-checking from typescript plugin to find more errors...

### CODE REVIEW

File: backend/match-reporter/MatchReporter.ts [hunk 1]
```diff
@@ -86,43 +94,46 @@ const createMatchResultMessage = (match, decorations) => {
...
-const createPurposeMessage = async rankings => {
+const createPurposeMessage = async (rankings): Promise<string> => {
   const rankingsText = rankings
     .map((ranking, i) => `${i + 1}. ${ranking.name} (${ranking.rating})`)
     .join('\n')
   const purpose = 'TOP PLAYERS\n' + rankingsText
   return purpose
 }
+
+export default class MatchReporter {
```
Comment (novellizator) [inline backend/match-reporter/MatchReporter.ts]: btw could you tun it into a named export?

Comment (littlewhywhat) [reply backend/match-reporter/MatchReporter.ts]: ok, sure! good idea

Comment (littlewhywhat) [reply backend/match-reporter/MatchReporter.ts]: Fixed - in https://github.com/salsita/foosball-rating/pull/103/commits/a62d62460c2477acf339cc077581a4520d1d5a29 and frontend errors in https://github.com/salsita/foosball-rating/pull/103/commits/cf98df2b8002545600f4dcefc45dc77deb957047

Comment (littlewhywhat) [reply backend/match-reporter/MatchReporter.ts]: I had an idea to put one command in root folder to lint both FE and BE but I had to move eslint dependencies to root package.json - made in https://github.com/salsita/foosball-rating/commit/87e3465bb1a304b7baf0ae876f5ae7e3c420e281 but not included to this PR yet. Maybe I would put it in a separate PR

Comment (novellizator) [reply backend/match-reporter/MatchReporter.ts]: both variant are fine with me

Comment (littlewhywhat) [reply backend/match-reporter/MatchReporter.ts]: Great, I will make a separate PR for https://github.com/salsita/foosball-rating/commit/87e3465bb1a304b7baf0ae876f5ae7e3c420e281 then

**Pull request merged**